packages/trpc/server/routers/viewer/workflows/util.ts

```diff
@@ -804,3 +807,85 @@ export async function getAllWorkflowsFromEventType(
 
   return allWorkflows;
 }
+
+export const getEventTypeWorkflows = async (
+  userId: number,
+  eventTypeId: number
+): Promise<z.infer<typeof ZWorkflows>> => {
+  const rawEventType = await prisma.eventType.findFirst({
+    where: {
+      AND: [
+        {
+          OR: [
+            {
+              users: {
+                some: {
+                  id: userId,
+                },
+              },
+            },
+            {
+              team: {
+                members: {
+                  some: {
+                    userId: userId,
+                  },
+                },
+              },
+            },
+            {
+              userId: userId,
+            },
+          ],
+        },
+        {
+          id: eventTypeId,
+        },
+      ],
+    },
```

Comment:
same logic as getEventTypeById. TODO: move this to repository pattern

Reply:
Why not put it in a repository now?

Reply:
the return type inside `getEventTypeById` is a bit complex and require too many relational fields data which is not required here and might slow down the result and we'll have to use Prisma.validator for all nested fields to make sure type checks pass

Reply:
let me try adding this to repository.